### PR TITLE
Refactor TimeSeries Component to simplify refresh

### PR DESCRIPTION
**Pull request by bthesorceror**
Closes application-team-issues#141

- Time range changes now reloads graph in flux editor

  - [x] Rebased/mergeable
  - [x] Tests pass
  - [x] Sign [CLA](https://influxdata.com/community/cla/) (if not already signed)

### CODE REVIEW

File: ui/package.json [hunk 1]
```diff
@@ -82,6 +82,7 @@
     "node-sass": "^4.9.0",
     "parcel": "^1.10.1",
     "prettier": "1.12.1",
+    "sass": "^1.14.2",
```
Comment (ischolten) [inline ui/package.json]: should this be a part of this pr?

File: ui/src/shared/components/time_series/TimeSeries.tsx [hunk 2]
```diff
@@ -128,59 +114,25 @@ class TimeSeries extends Component<Props, State> {
...
   public async componentDidUpdate(prevProps: Props) {
-    if (this.props.autoRefresher !== prevProps.autoRefresher) {
-      prevProps.autoRefresher.unsubscribe(this.executeQueries)
-      this.props.autoRefresher.subscribe(this.executeQueries)
-    }
-
-    if (!this.isPropsDifferent(prevProps)) {
-      return
+    const prevQueries = _.map(prevProps.queries, q => _.pick(q, ['text']))
+    const currQueries = _.map(this.props.queries, q => _.pick(q, ['text']))
```
Comment (ischolten) [inline ui/src/shared/components/time_series/TimeSeries.tsx]: could these not just be `q => q.text` ?

File: ui/src/shared/components/AutoRefresh.tsx [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+interface State {
+  id: string
+  manualRefresh: number
+}
+
+export default class AutoRefresh extends PureComponent<Props, State> {
+  public static getDerivedStateFromProps(nextProps: Props, state: State) {
+    if (state.manualRefresh !== nextProps.manualRefresh) {
+      return {
+        id: uuid.v4(),
```
[the remark this answers is not in the document]
Comment (bthesorceror) [reply ui/src/shared/components/AutoRefresh.tsx]: manualRefresh is the name we use through the application to indicate the timestamp of when you hit the manual refresh button. I would prefer to leave the name the same in this PR and if we need to rename it, then rename it in all the places.

File: ui/src/shared/components/AutoRefresh.tsx [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+  public componentWillUnmount() {
+    const {autorefresh} = this.props
+
+    if (autorefresh) {
+      autorefresh.unsubscribe(this.update)
+    }
+  }
+
+  private update = () => {
```
[the remark this answers is not in the document]
Comment (bthesorceror) [reply ui/src/shared/components/AutoRefresh.tsx]: The name is to imply that it updates the component from the autorefresher singleton

[the remark this answers is not in the document]
Comment (bthesorceror) [reply ui/src/shared/components/AutoRefresh.tsx]: updating the id is an implementation detail

File: ui/src/shared/components/AutoRefresh.tsx [hunk 5]
```diff
@@ -0,0 +1,60 @@
+import uuid from 'uuid'
+import {PureComponent} from 'react'
+
+import {AutoRefresher} from 'src/utils/AutoRefresher'
+
+interface Props {
+  autorefresh?: AutoRefresher
```
[the remark this answers is not in the document]
Comment (bthesorceror) [reply ui/src/shared/components/AutoRefresh.tsx]: 👍

File: ui/src/shared/components/AutoRefresh.tsx [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+export default class AutoRefresh extends PureComponent<Props, State> {
+  public static getDerivedStateFromProps(nextProps: Props, state: State) {
+    if (state.manualRefresh !== nextProps.manualRefresh) {
+      return {
+        id: uuid.v4(),
+        manualRefresh: nextProps.manualRefresh,
+      }
+    }
+
+    return false
```
[the remark this answers is not in the document]
Comment (bthesorceror) [reply ui/src/shared/components/AutoRefresh.tsx]: 👍

File: ui/src/shared/components/AutoRefresh.tsx [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+}
+
+interface State {
+  id: string
+  manualRefresh: number
+}
+
+export default class AutoRefresh extends PureComponent<Props, State> {
+  public static getDerivedStateFromProps(nextProps: Props, state: State) {
+    if (state.manualRefresh !== nextProps.manualRefresh) {
```
[the remark this answers is not in the document]
Comment (bthesorceror) [reply ui/src/shared/components/AutoRefresh.tsx]: see comment below

File: ui/src/kapacitor/components/RuleGraph.tsx [hunk 8]
```diff
@@ -56,6 +57,7 @@ class RuleGraph extends PureComponent<Props> {
             timeRange={timeRange}
             source={source}
             queries={this.queries}
+            uuid={uuid.v1()}
```
[the remark this answers is not in the document]
Comment (bthesorceror) [reply ui/src/kapacitor/components/RuleGraph.tsx]: 👍

File: ui/src/shared/components/RefreshingGraph.tsx [hunk 9]
```diff
@@ -144,53 +140,100 @@ class RefreshingGraph extends PureComponent<Props> {
...
+                  const v = _.some(results, r => r.series)
+                  return v
+                })
+
+              if (!hasValues) {
+                if (cellNoteVisibility === NoteVisibility.ShowWhenNoData) {
+                  return <MarkdownCell text={cellNote} />
+                }
+
+                if (this.isFluxQuery && !getDeep(source, 'links.flux', null)) {
```
[the remark this answers is not in the document]
Comment (bthesorceror) [reply ui/src/shared/components/RefreshingGraph.tsx]: 👍

File: ui/src/shared/components/AutoRefresh.tsx [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+  }
+
+  public render() {
+    return this.props.children(this.state.id)
+  }
+
+  public componentDidMount() {
+    const {autorefresh} = this.props
+
+    if (autorefresh) {
```
[the remark this answers is not in the document]
Comment (bthesorceror) [reply ui/src/shared/components/AutoRefresh.tsx]: autoRefresh could be null in some situations

**Pull request merged**